src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -1749,7 +1750,9 @@ int32_t SystemNative_GetSockOpt(
                 value = value == 0 ? 1 : 0;
             }
             *(int32_t*)optionValue = value;
-
+#else
+            return Error_ENOTSUP;
```

Comment:
This will result in an exception getting thrown.  How does Mono currently react to Socket.ExclusiveAddressUse being set if the platform can't support it?  Should we only throw if it's set to a specific value? Etc.